sljit_src/sljitNativeS390X.c

```diff
@@ -0,0 +1,2817 @@
+				sr(src1_r, src2_r):
+				slr(src1_r, src2_r);
+			break;
+		case SLJIT_ADDC32:
+			ins = alcr(src1_r, src2_r);
+			break;
+		case SLJIT_SUBC32:
+			ins = slbr(src1_r, src2_r);
+			break;
+		case SLJIT_MUL32:
+			ins = msr(src1_r, src2_r);
+			break;
+		case SLJIT_AND32:
+			ins = nr(src1_r, src2_r);
+			break;
+		case SLJIT_OR32:
+			ins = or(src1_r, src2_r);
+			break;
+		case SLJIT_XOR32:
+			ins = xr(src1_r, src2_r);
+			break;
+		default:
+			SLJIT_UNREACHABLE();
+		}
+		FAIL_IF(push_inst(compiler, ins));
+		if (src1_r != dst_r)
+			FAIL_IF(push_inst(compiler, (op & SLJIT_I32_OP) ?
+				lr(dst_r, src1_r) : lgr(dst_r, src1_r)));
+
+	}
+
+	/* write condition code to emulated flag register */
+	if (op & VARIABLE_FLAG_MASK)
+		FAIL_IF(push_inst(compiler, ipm(flag_r)));
+
+	/* write zero flag to emulated flag register */
+	if (op & SLJIT_SET_Z)
+		FAIL_IF(push_store_zero_flag(compiler, op, dst_r));
+
+	/* finally write the result to memory if required */
+	if (dst & SLJIT_MEM) {
+		SLJIT_ASSERT(dst_r != tmp1);
+		/* TODO(carenas): s/FAIL_IF/ return */
+		FAIL_IF(store_word(compiler, dst_r, dst, dstw, tmp1, op & SLJIT_I32_OP));
+	}
+
+	return SLJIT_SUCCESS;
+}
+
+SLJIT_API_FUNC_ATTRIBUTE sljit_s32 sljit_emit_op_src(
+	struct sljit_compiler *compiler,
+	sljit_s32 op, sljit_s32 src, sljit_sw srcw)
+{
+	sljit_gpr src_r;
+
+	CHECK_ERROR();
+	CHECK(check_sljit_emit_op_src(compiler, op, src, srcw));
+	ADJUST_LOCAL_OFFSET(src, srcw);
+
+	switch (op) {
+	case SLJIT_FAST_RETURN:
+		src_r = FAST_IS_REG(src) ? gpr(src) : tmp1;
+		if (src & SLJIT_MEM)
+			FAIL_IF(load_word(compiler, tmp1, src, srcw, tmp1, 0));
+
+		return push_inst(compiler, br(src_r));
+	case SLJIT_SKIP_FRAMES_BEFORE_FAST_RETURN:
+	case SLJIT_PREFETCH_L1:
+	case SLJIT_PREFETCH_L2:
+	case SLJIT_PREFETCH_L3:
+	case SLJIT_PREFETCH_ONCE:
+	default:
+		return SLJIT_SUCCESS;
+	}
+
+	return SLJIT_SUCCESS;
+}
+
+SLJIT_API_FUNC_ATTRIBUTE sljit_s32 sljit_get_register_index(sljit_s32 reg)
+{
+	CHECK_REG_INDEX(check_sljit_get_register_index(reg));
+	return gpr(reg);
+}
+
+SLJIT_API_FUNC_ATTRIBUTE sljit_s32 sljit_get_float_register_index(sljit_s32 reg)
+{
+	CHECK_REG_INDEX(check_sljit_get_float_register_index(reg));
+	abort();
+}
+
+SLJIT_API_FUNC_ATTRIBUTE sljit_s32 sljit_emit_op_custom(struct sljit_compiler *compiler,
+	void *instruction, sljit_s32 size)
+{
+	sljit_u8 *src = (sljit_u8*)instruction;
+	sljit_ins ins = 0;
+
+	CHECK_ERROR();
+	CHECK(check_sljit_emit_op_custom(compiler, instruction, size));
+
+	switch (size) {
+	case 2:
+		ins |= (sljit_ins)(src[0]) << 8;
+		ins |= (sljit_ins)(src[1]);
+		break;
+	case 4:
+		ins |= (sljit_ins)(src[0]) << 24;
+		ins |= (sljit_ins)(src[1]) << 16;
+		ins |= (sljit_ins)(src[2]) << 8;
+		ins |= (sljit_ins)(src[3]);
+		break;
+	case 6:
+		ins |= (sljit_ins)(src[0]) << 40;
+		ins |= (sljit_ins)(src[1]) << 32;
+		ins |= (sljit_ins)(src[2]) << 24;
+		ins |= (sljit_ins)(src[3]) << 16;
+		ins |= (sljit_ins)(src[4]) << 8;
+		ins |= (sljit_ins)(src[5]);
```

Comment:
What about using an `memcpy((sljit_u8*)&inst + sizeof(sljit_ins) - size, instruction, size);` instead of the switch statement? This should work because the system is big endian, and the inst is initialized to 0.